### DISCUSSION TITLE
Can we put case switching under match mode?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#3818] [Ideas]
Helix uses:
    * `tilde` to switch to lowercase
    * `alt+tilde` to switch to uppercase
    * `~` to change case

We could put these under match mode so that:
    * `ml`: make lowercase
    * `mu`: make uppercase
    * `mt`: switch case(`t` for transform/transition...?)

**Comment by sudormrfbin** [comment]
It certainly makes it easier to type, but it doesn't make much sense to put it in _match_ mode - it has keybinds to match around and inside text (textobjects), delimiters (surround), etc. But there's nothing "matchy" about swapping case so at least as far as the name of the mode goes it seems to be a bad fit.

**Reply by txtyash** [reply]
Yes, but typing convenience should have more preference. For naming's sake, the name of the match mode can be changed to something else or there can be two different mode names under the `m` key.

Also, `make uppercase` & `make lowercase` sound proper.

Simply renaming `match` mode to `match, make` mode?

**Reply by dead10ck** [reply]
> Yes, but typing convenience should have more preference.

This is your view. It's a perfectly legitimate one, and Helix gives you the ability to do this for yourself. But this project explicitly favors consistency over the optimal minimum of finger movement.

Configuration exists to let people choose how they want their editor to behave. I suggest you spend this effort on your own config.
